syntax = "proto3";

package zorro;
option go_package = "github.com/Acedyn/zorro-proto/zorroprotos/scheduling";

import "zorroprotos/processor/processor.proto";

// Parameters for processor registration
message ProcessorRegistration {
  Processor processor = 1;
  string host = 2;
}

service Scheduling {
  // For processors to start running their commands
  rpc RegisterProcessor(ProcessorRegistration) returns (Processor);
}
